test/e2e/storage_test.go

```diff
@@ -0,0 +1,60 @@
+package e2e
+
+import (
+	"context"
+	"testing"
+
+	config "github.com/openshift/api/config/v1"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	core "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	meta "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	azureProvider "github.com/openshift/windows-machine-config-operator/test/e2e/providers/azure"
+)
+
+// TestStorage tests that persistent storage can be used with Windows workloads on supported providers
+func TestStorage(t *testing.T) {
+	tc, err := NewTestContext()
+	require.NoError(t, err)
+	// If more cloud provider storage tests are to be added, the code which provisions StorageClasses and PVCs should
+	// be moved into the CloudProvider interface
+	if tc.CloudProvider.GetType() != config.AzurePlatformType {
+		t.Skip("storage not supported on this platform")
+
+	}
+	provider, isAzureProvider := tc.CloudProvider.(*azureProvider.Provider)
+	require.True(t, isAzureProvider)
+	scSpec, err := provider.GenerateStorageClass()
+	require.NoError(t, err)
+	sc, err := tc.client.K8s.StorageV1().StorageClasses().Create(context.TODO(), scSpec, meta.CreateOptions{})
+	require.NoError(t, err)
+	defer tc.client.K8s.StorageV1().StorageClasses().Delete(context.TODO(), sc.GetName(), meta.DeleteOptions{})
+	pvcSpec := core.PersistentVolumeClaim{
+		ObjectMeta: meta.ObjectMeta{
+			Name: "storage",
+		},
+		Spec: core.PersistentVolumeClaimSpec{
+			AccessModes: []core.PersistentVolumeAccessMode{core.ReadWriteMany},
+			Resources: core.ResourceRequirements{
+				Requests: core.ResourceList{core.ResourceStorage: resource.MustParse("2Gi")},
+			},
+			StorageClassName: &sc.Name,
+		},
+	}
+	pvc, err := tc.client.K8s.CoreV1().PersistentVolumeClaims(tc.workloadNamespace).Create(context.TODO(), &pvcSpec,
+		meta.CreateOptions{})
+	require.NoError(t, err)
+	defer tc.client.K8s.CoreV1().PersistentVolumeClaims(tc.workloadNamespace).Delete(context.TODO(), pvc.GetName(),
+		meta.DeleteOptions{})
+
+	// Testing that the pod goes to ready is sufficient as the pod will not become ready without successfully mounting
+	// the volume
+	ws, err := tc.deployWindowsWebServer("storage-test", nil,
```

Comment:
So we are deploying yet another webserver to test storage. Can't we deploy the webserver before we run the networking and storage tests and use them in both? I am worried about the increase in time this is going to take. I realize this might take a refactor but please see if it is easily doable. We are going to have to do this anyhow as we keep adding more storage support for more providers.